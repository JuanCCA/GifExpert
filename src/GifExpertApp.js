import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defuaultCategories = []}) => {

   // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(defuaultCategories);

    const handleAdd = () =>{
       //setCategories( [...categories, 'Hero Academia'])
       
       setCategories( (categ) => [...categories, 'Hero Academia'])
    }
    
    return (
        <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories}/>
        <hr />

        <button onClick={handleAdd}>AÃ±adir</button>

        <ol>
            {
                categories.map( category => (
                    <GifGrid category={ category }
                             key={ category }
                    />
                ))
            }
        </ol>
        </>
    )
    
    
}




export default GifExpertApp

