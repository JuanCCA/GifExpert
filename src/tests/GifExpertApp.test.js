import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp'
import React from 'react';

describe('Testing GifExpertApp', () => {
    

    test('should create component correctly', () => {
        
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();


    })

    test('should create component with categories', () => {
        
        const list = ["a","b"]

        const wrapper = shallow(<GifExpertApp defuaultCategories={list} />);


        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(list.length)


    })
    


})
