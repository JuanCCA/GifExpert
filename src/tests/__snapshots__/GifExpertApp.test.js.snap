// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp should create component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <button
    onClick={[Function]}
  >
    Añadir
  </button>
  <ol />
</Fragment>
`;

exports[`Testing GifExpertApp should create component with categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <button
    onClick={[Function]}
  >
    Añadir
  </button>
  <ol>
    <GifGrid
      category="a"
      key="a"
    />
    <GifGrid
      category="b"
      key="b"
    />
  </ol>
</Fragment>
`;
